diff --git a/Source/Tools/FEXInterpreter/FEXInterpreter.cpp b/Source/Tools/FEXInterpreter/FEXInterpreter.cpp
index 3335531..b3162c7 100644
--- a/Source/Tools/FEXInterpreter/FEXInterpreter.cpp
+++ b/Source/Tools/FEXInterpreter/FEXInterpreter.cpp
@@ -270,7 +270,7 @@ namespace TSO {
     if (Result == PR_SET_MEM_MODEL_DEFAULT) {
       // Try to set the TSO mode if we are currently default.
       Result = prctl(PR_SET_MEM_MODEL, PR_SET_MEM_MODEL_TSO, 0, 0, 0);
-      if (Result == 0) {
+      if (Result == 0 && !getenv("FEX_DISABLE_HARDWARE_TSO")) {
         // TSO mode successfully enabled. Tell the context to disable TSO emulation through atomics.
         // This flag gets inherited on thread creation, so FEX only needs to set it at the start.
         CTX->SetHardwareTSOSupport(true);
