--- a/nix/vendored-packages.nix
+++ b/nix/vendored-packages.nix
@@ -17,6 +17,7 @@
       format = "wheel";
       src = pkgs.fetchurl { inherit url hash; };
       doCheck = false;
+      dontCheckRuntimeDeps = true;
     };
 
   workflowTemplatesCore = mkWheel {
@@ -115,6 +116,7 @@
       inherit pname version;
       format = "wheel";
       src = pkgs.fetchurl { inherit url hash; };
+      dontCheckRuntimeDeps = true;
       doCheck = false;
       nativeBuildInputs = pkgs.lib.optionals pkgs.stdenv.isLinux [ pkgs.autoPatchelfHook ];
       buildInputs = pkgs.lib.optionals pkgs.stdenv.isLinux [ pkgs.stdenv.cc.cc.lib ];
